contract-examples/test/contract_deployer_allow_list.ts

```diff
@@ -1,117 +1,50 @@
 // (c) 2019-2022, Ava Labs, Inc. All rights reserved.
 // See the file LICENSE for licensing terms.
 
-import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
-import { expect } from "chai";
-import {
-  Contract,
-  ContractFactory,
-} from "ethers"
 import { ethers } from "hardhat"
+import { test } from "./utils"
 
-// make sure this is always an admin for minter precompile
-const adminAddress: string = "0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"
-const ALLOWLIST_ADDRESS = "0x0200000000000000000000000000000000000000";
-
-const ROLES = {
-  NONE: 0,
-  DEPLOYER: 1,
-  ADMIN: 2
-};
+const ADMIN_ADDRESS: string = "0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"
+const OTHER_SIGNER = "0x0Fa8EA536Be85F32724D57A37758761B86416123"
+const DEPLOYER_ALLOWLIST_ADDRESS = "0x0200000000000000000000000000000000000000"
 
 describe("ExampleDeployerList", function () {
-  let owner: SignerWithAddress
-  let contract: Contract
-  let deployer: SignerWithAddress
-  before(async function () {
-    owner = await ethers.getSigner(adminAddress);
-    const contractF: ContractFactory = await ethers.getContractFactory("ExampleDeployerList", { signer: owner })
-    contract = await contractF.deploy()
-    await contract.deployed()
-    const contractAddress: string = contract.address
-    console.log(`Contract deployed to: ${contractAddress}`)
-
-    const signers: SignerWithAddress[] = await ethers.getSigners()
-    deployer = signers.slice(-1)[0]
-
-    // Fund deployer address
-    await owner.sendTransaction({
-      to: deployer.address,
-      value: ethers.utils.parseEther("1")
-    })
-
-  });
+  beforeEach('Setup DS-Test contract', async function () {
+    const signer = await ethers.getSigner(ADMIN_ADDRESS)
+    const allowListPromise = ethers.getContractAt("IAllowList", DEPLOYER_ALLOWLIST_ADDRESS, signer)
 
-  it("precompile should see owner address has admin role", async function () {
-    // test precompile first
-    const allowList = await ethers.getContractAt("IAllowList", ALLOWLIST_ADDRESS, owner);
-    let adminRole = await allowList.readAllowList(owner.address);
-    expect(adminRole).to.be.equal(ROLES.ADMIN)
-  });
+    return ethers.getContractFactory("ExampleDeployerListTest", { signer })
```

Comment:
what happens to the return value of `beforeEach`? (is the result of  tx.wait() checked or used somewhere?)

Reply:
Nothing happens to the return value, you need to either return or await the promise. `return` is more robust as you can convert the function to a regular function (instead of `async`) and it will behave the same way.

Reply:
Should we await so we can assert the tx's receipt status?

Reply:
I don't understand, the next beforeEach block or test function won't be triggered until that final `tx.wait()` resolves below.